morpheus/_lib/src/io/deserializers.cpp

```diff
@@ -111,23 +111,23 @@ cudf::io::table_with_metadata load_table_from_file(const std::string& filename)
 int get_index_col_count(cudf::io::table_with_metadata& data_table)
```

Comment:
Dont love that this mutates the input argument. We should make a second method called `prepare_df_index` or something similar that does the index rename and also returns the index count. Then you can call `int get_index_col_count(const cudf::io::table_with_metadata& data_table)` if you just need the index count and `int prepare_df_index(cudf::io::table_with_metadata& data_table)` if you need to get the count and update the column name.

Reply:
done